floris/optimization/load_optimization/load_optimization.py

```diff
@@ -0,0 +1,463 @@
+
+    # Get the ambient wind speed and apply to each turbine per findex
+    ambient_wind_speeds = fmodel.wind_speeds
+    ambient_wind_speeds = np.tile(ambient_wind_speeds[:, np.newaxis], (1, fmodel.n_turbines))
+
+    # Compute the rotor area
+    D = fmodel.core.farm.rotor_diameters[0, 0]
+    area = np.pi * (D / 2) ** 2
+
+    # Compute the thrust
+    cts = fmodel.get_turbine_thrust_coefficients()
+    thrust = 0.5 * fmodel.core.flow_field.air_density * area * cts * ambient_wind_speeds**2
+
+    # Compute the load_ti
+    load_ti = compute_load_ti(
+        fmodel=fmodel,
+        load_ambient_tis=load_ambient_tis,
+        wake_slope=wake_slope,
+        max_dist_D=max_dist_D,
+    )
+
+    # Compute wind speed standard deviation
+    ws_std = ambient_wind_speeds * load_ti
+
+    # Compute voc
+    return A * (ws_std**exp_ws_std) * (thrust**exp_thrust)
+
+
+def compute_farm_voc(
+    fmodel: FlorisModel,
+    A: float,
+    load_ambient_tis: np.array,
+    wake_slope: float = 0.3,
+    max_dist_D: float = 10.0,
+    exp_ws_std: float = 1.0,
+    exp_thrust: float = 1.0,
+):
+    """Compute the farm Variable Operating Cost (VOC) for each findex.
+
+    Args:
+        fmodel (FlorisModel): FlorisModel object
+        A (float): Coefficient for the VOC calculation
+        load_ambient_tis (list or np.array): Ambient 'load' turbulence intensity for each findex
+        wake_slope (float, optional): Wake slope. Defaults to 0.3.
+        max_dist_D (flat, optional): Maximum distance in rotor diameters. Defaults to 10.0.
+        exp_ws_std (float, optional): Exponent for the wind speed standard deviation.
+            Defaults to 1.0.
+        exp_thrust (float, optional): Exponent for the thrust. Defaults to 1.0.
+
+    Returns:
+        np.array: Array of farm VOC for each findex
+
+    """
+    turbine_voc = compute_turbine_voc(
+        fmodel=fmodel,
+        A=A,
+        load_ambient_tis=load_ambient_tis,
+        wake_slope=wake_slope,
+        max_dist_D=max_dist_D,
+        exp_ws_std=exp_ws_std,
+        exp_thrust=exp_thrust,
+    )
+    return np.sum(turbine_voc, axis=1)
+
+
+def compute_farm_revenue(
+    fmodel: FlorisModel,
+):
+    """Compute the farm revenue of the FlorisModel object.
+
+    Args:
+        fmodel (FlorisModel): FlorisModel object
+
+    Returns:
+        np.array: Array of farm revenue for each findex
+
+    """
+
+    if fmodel.core.state is not State.USED:
+        raise ValueError("FlorisModel must be run before computing net revenue")
+
+    # Make sure fmodel.wind_data is not None
+    if fmodel.wind_data is None:
+        raise ValueError("FlorisModel must have wind_data to compute net revenue")
+
+    # Ensure that fmodel.wind_data is not None
+    if fmodel.wind_data.values is None:
+        raise ValueError("FlorisModel wind_data.values must be set to compute revenue")
+
+    farm_power = fmodel.get_farm_power()
+    values = fmodel.wind_data.values
+    return farm_power * values
```

Comment:
Should there be some sort of dt in here? How does that work? Or is the output revenue treated as "per unit time", so to speak?

Reply:
Could we just document that values are interpreted as revenue per Watt per findex? Same with VOC? Cost per  findex?

Reply:
We should probably just try to make this clear in the examples since this is just re-using the underlying values architecture